unipop-elastic/src/org/unipop/elastic/common/QueryIterator.java

```diff
@@ -1,65 +1,96 @@
+    private JestResult scrollResponse;
+    private String scrollId;
     private final Parser<E> parser;
     private final int scrollSize;
-    private Client client;
-    private Iterator<SearchHit> hits;
+    private JestClient client;
+    private Iterator<JsonElement> hits;
+    private final Gson gson;
 
-    public QueryIterator(QueryBuilder query, StepDescriptor stepDescriptor, int scrollSize, int maxSize, Client client,
-                         Parser<E> parser, String... indices) {
-        this.stepDescriptor = stepDescriptor;
+    public QueryIterator(QueryBuilder query, int scrollSize, int maxSize, JestClient client,
+                         Parser<E> parser, String... indices) throws IOException {
         this.scrollSize = scrollSize;
         this.client = client;
         this.parser = parser;
 
-        SearchRequestBuilder searchRequestBuilder = client.prepareSearch(indices).setQuery(query);
+        this.gson = new Gson();
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
+                .query(query);
+
+        Search.Builder searchBuilder = new Search.Builder(searchSourceBuilder.toString())
+                .addIndex(Seq.of(indices).toList());
+
         if(maxSize == -1) maxSize = 10000;
 
         if (scrollSize > 0) {
-            searchRequestBuilder.setScroll(new TimeValue(60000))
-                    .setSize(maxSize < scrollSize ? maxSize : scrollSize);
+            searchBuilder.setParameter(Parameters.SCROLL, new TimeValue(60000));
+            searchBuilder.setParameter(Parameters.SIZE, Math.min(maxSize, scrollSize));
+        }
+
+        else {
+            searchBuilder.setParameter(Parameters.SIZE, maxSize);
         }
-        else searchRequestBuilder.setSize(maxSize);
 
-        this.scrollResponse = searchRequestBuilder.execute().actionGet();
+        Search search = searchBuilder.build();
 
-        hits = scrollResponse.getHits().iterator();
+        this.scrollResponse = client.execute(search);
+        this.hits = this.scrollResponse.getJsonObject().getAsJsonObject("hits").getAsJsonArray("hits").iterator();
     }
 
     @Override
     public boolean hasNext() {
         if (hits.hasNext()) return true;
 
         if (scrollSize > 0) {
-            scrollResponse = client.prepareSearchScroll(scrollResponse.getScrollId()).setScroll(new TimeValue(600000)).execute().actionGet();
-            hits = scrollResponse.getHits().iterator();
+            try {
+                SearchScroll scroll = new SearchScroll.Builder(
+                        this.scrollResponse.getJsonObject().getAsJsonPrimitive("_scroll_id").getAsString(),
+                        new TimeValue(60000).format())
+                        .build();
+                this.scrollResponse = client.execute(scroll);
+                this.hits = scrollResponse.getJsonObject().getAsJsonObject("hits").getAsJsonArray("hits").iterator();
+            } catch (IOException e) {
+                // TODO: Handle Exception
```

Comment:
Yea, but meanwhile do atleast System.out.println.. Or else if there's a problem we won't know about it

Get Outlook for iOS




On Sun, Jun 19, 2016 at 8:03 PM +0300, "Gur Ronen" <notifications@github.com> wrote:












In unipop-elastic/src/org/unipop/elastic/common/QueryIterator.java:
>      }
>  
>      @Override
>      public boolean hasNext() {
>          if (hits.hasNext()) return true;
>  
>          if (scrollSize > 0) {
> -            scrollResponse = client.prepareSearchScroll(scrollResponse.getScrollId()).setScroll(new TimeValue(600000)).execute().actionGet();
> -            hits = scrollResponse.getHits().iterator();
> +            try {
> +                SearchScroll scroll = new SearchScroll.Builder(
> +                        this.scrollResponse.getJsonObject().getAsJsonPrimitive("_scroll_id").getAsString(),
> +                        new TimeValue(60000).format())
> +                        .build();
> +                this.scrollResponse = client.execute(scroll);
> +                this.hits = scrollResponse.getJsonObject().getAsJsonObject("hits").getAsJsonArray("hits").iterator();
> +            } catch (IOException e) {
> +                // TODO: Handle Exception



left this as a todo until we decide what to do about logging



—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub, or mute the thread.